### PR TITLE
Adds testing for CL_PROGRAM_IL

### LINKED ISSUE #164: CL_PROGRAM_IL not being tested
Testing of `clGetDeviceInfo` with `param_name` of `CL_PROGRAM_IL` appears to be completely missing.

### PR DESCRIPTION
**Pull request by ellnor01**
Added a test to check SpirV as an IL for programs built with IL
which includes a negative test for programs built without IL.

Added a test_fail macro.

Fixes #164

Signed-off-by: Ellen Norris-Thompson <ellen.norris-thompson@arm.com>

### CODE REVIEW

File: test_common/harness/errorHelpers.h [hunk 1]
```diff
@@ -60,6 +60,11 @@ extern "C" {
 #define ct_assert_i(b, line)  ct_assert_ii(b, line)
 #define ct_assert_ii(b, line) int _compile_time_assertion_on_line_##line[b ? 1 : -1];
 
+#define test_fail(msg, ...)                                                    \
```
Comment (bashbaug) [inline test_common/harness/errorHelpers.h:63]: Could this be superseded by `test_assert_error` in #912, which adds similar functionality?

File: test_conformance/spirv_new/test_get_program_il.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    "    dst[tid] = (int)src[tid];\n"
+    "\n"
+    "}\n"
+};
+
+TEST_SPIRV_FUNC(get_program_il)
+{
+    cl_platform_id platform;
+    clProgramWrapper source_program;
+    Version device_version = get_device_cl_version(deviceID);
```
Comment (bashbaug) [inline test_conformance/spirv_new/test_get_program_il.cpp:32]: Minor comment: It looks like `platform` and `device_version` are unused.

File: test_common/harness/errorHelpers.h (hunk 1 above)
Comment (ellnor01) [reply test_common/harness/errorHelpers.h:63]: I have another PR to upload for a different issue which also uses this define. Would you mind if I left this for now and I'll go back and swap all instances for test_assert_error once the PRs are merged?

**Comment by ellnor01** [comment]
> Also, is a change to the CMakelists needed? I tried building this PR but it didn't build the new file.

I believe the CMakelists.txt for spirv_new collects all the .cpp files within the directory to build, they are not named individually.

File: test_common/harness/errorHelpers.h (hunk 1 above)
Comment (bashbaug) [reply test_common/harness/errorHelpers.h:63]: Works for me!

**Comment by bashbaug** [comment]
> > Also, is a change to the CMakelists needed? I tried building this PR but it didn't build the new file.
> 
> I believe the CMakelists.txt for spirv_new collects all the .cpp files within the directory to build, they are not named individually.

Ah, yes, this is correct.  I did have to manually re-run CMake, but after doing this I was able to build and run your test.  Thanks!

**Review by kpet — APPROVED** [review]

**Pull request merged**